core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageAutoFeatureStep.java

```diff
@@ -80,6 +82,22 @@ public void write(String s, byte[] bytes) {
         TryBlock overallCatch = beforeAn.tryBlock();
         //TODO: at some point we are going to need to break this up, as if it get too big it will hit the method size limit
 
+        ResultHandle beforeAnalysisParam = beforeAn.getMethodParam(0);
+        for (UnsafeAccessedFieldBuildItem unsafeAccessedField : unsafeAccessedFields) {
+            TryBlock tc = overallCatch.tryBlock();
+            ResultHandle declaringClassHandle = tc.invokeStaticMethod(
+                    ofMethod(Class.class, "forName", Class.class, String.class),
+                    tc.load(unsafeAccessedField.getDeclaringClass()));
+            ResultHandle fieldHandle = tc.invokeVirtualMethod(
+                    ofMethod(Class.class, "getDeclaredField", Field.class, String.class), declaringClassHandle,
+                    tc.load(unsafeAccessedField.getFieldName()));
+            tc.invokeInterfaceMethod(
+                    ofMethod(Feature.BeforeAnalysisAccess.class, "registerAsUnsafeAccessed", void.class, Field.class),
+                    beforeAnalysisParam, fieldHandle);
+            CatchBlockCreator cc = tc.addCatch(Throwable.class);
+            cc.invokeVirtualMethod(ofMethod(Throwable.class, "printStackTrace", void.class), cc.getCaughtException());
```

Comment:
Hmmm, we don't use a proper logger here?

Reply:
I don't know why, but this is how it's done in the entire class.